redis_timeseries.go

```diff
@@ -0,0 +1,877 @@
+				args = append(args, label, value)
+
+			}
+		}
+	}
+	cmd := NewStatusCmd(ctx, args...)
+	_ = c(ctx, cmd)
+	return cmd
+}
+
+// TSCreateRule - Creates a compaction rule from sourceKey to destKey.
+// For more information - https://redis.io/commands/ts.createrule/
+func (c cmdable) TSCreateRule(ctx context.Context, sourceKey string, destKey string, aggregator string, bucketDuration int) *StatusCmd {
+	args := []interface{}{"TS.CREATERULE", sourceKey, destKey, "AGGREGATION", aggregator, bucketDuration}
+	cmd := NewStatusCmd(ctx, args...)
+	_ = c(ctx, cmd)
+	return cmd
+}
+
+// TSCreateRuleArgs - Creates a compaction rule from sourceKey to destKey with additional option.
+// This function allows for specifying additional option such as:
+// alignTimestamp.
+// For more information - https://redis.io/commands/ts.createrule/
+func (c cmdable) TSCreateRuleArgs(ctx context.Context, sourceKey string, destKey string, aggregator string, bucketDuration int, options *TSCreateRuleOptions) *StatusCmd {
+	args := []interface{}{"TS.CREATERULE", sourceKey, destKey, "AGGREGATION", aggregator, bucketDuration}
+	if options != nil {
+		if options.alignTimestamp != 0 {
+			args = append(args, options.alignTimestamp)
+		}
+	}
+	cmd := NewStatusCmd(ctx, args...)
+	_ = c(ctx, cmd)
+	return cmd
+}
+
+// TSIncrBy - Increments the value of a time-series key by the specified addend.
+// For more information - https://redis.io/commands/ts.incrby/
+func (c cmdable) TSIncrBy(ctx context.Context, Key string, addend float64) *IntCmd {
+	args := []interface{}{"TS.INCRBY", Key, addend}
+	cmd := NewIntCmd(ctx, args...)
+	_ = c(ctx, cmd)
+	return cmd
+}
+
+// TSIncrByArgs - Increments the value of a time-series key by the specified addend with additional options.
+// This function allows for specifying additional options such as:
+// Timestamp, Retention, ChunkSize, Uncompressed and Labels.
+// For more information - https://redis.io/commands/ts.incrby/
+func (c cmdable) TSIncrByArgs(ctx context.Context, key string, addend float64, options *TSIncrDecrOptions) *IntCmd {
+	args := []interface{}{"TS.INCRBY", key, addend}
+	if options != nil {
+		if options.Timestamp != 0 {
+			args = append(args, "TIMESTAMP", options.Timestamp)
+		}
+		if options.Retention != 0 {
+			args = append(args, "RETENTION", options.Retention)
+		}
+		if options.ChunkSize != 0 {
+			args = append(args, "CHUNK_SIZE", options.ChunkSize)
+
+		}
+		if options.Uncompressed {
+			args = append(args, "UNCOMPRESSED")
+		}
+		if options.Labels != nil {
+			args = append(args, "LABELS")
+			for label, value := range options.Labels {
+				args = append(args, label, value)
+
+			}
+		}
+	}
+	cmd := NewIntCmd(ctx, args...)
+	_ = c(ctx, cmd)
+	return cmd
+}
+
+// TSDecrBy - Decrements the value of a time-series key by the specified subtrahend.
+// For more information - https://redis.io/commands/ts.decrby/
+func (c cmdable) TSDecrBy(ctx context.Context, Key string, subtrahend float64) *IntCmd {
```

Comment:
I know - the docs call this variable subtrahend (and addend in TS.INCRBY), IMHO this isn't clear.  Maybe we should just use 'timestamp' as a variable name.